dbt/adapters/spark/relation.py

```diff
@@ -27,12 +33,37 @@ class SparkRelation(BaseRelation):
     quote_character: str = "`"
     is_delta: Optional[bool] = None
     is_hudi: Optional[bool] = None
+    is_iceberg: Optional[bool] = None
     information: Optional[str] = None
+    loader: Optional[str] = None
+    source_meta: Optional[Dict[str, Any]] = None
+    meta: Optional[Dict[str, Any]] = None
 
     def __post_init__(self):
-        if self.database != self.schema and self.database:
+        if self.is_iceberg is not True and self.database != self.schema and self.database:
```

Comment:
This seems like maybe it should be `or`? Any of these conditions are failures or otherwise need to be checked everywhere.

If this is only the post_init that seems fine to me, but this feels like two independent pieces of information (is qualified / namespaces relation and then the is iceberg). Is that not correct?

Reply:
I'll give that change a try in my testcases.  I think you're right here.

Reply:
Alright so using or's in this case quickly made all the tox tests fail :).  Likely because the first condition of the if is always going to evaluate to true for the base tests and none of them are for iceberg.